pkg/ebpf/tracee.go

```diff
@@ -240,6 +240,30 @@ func New(cfg Config) (*Tracee, error) {
 		}
 	}
 
+	// Clean up signatures not chosen for tracing
+	// Looping from the back to avoid bounds checking
+	for i := len(t.config.EngineConfig.Signatures) - 1; i > 0; i-- {
+		sig := t.config.EngineConfig.Signatures[i]
+		md, _ := sig.GetMetadata()
+		sigEventName := md.EventName
+		sigEventId, ok := events.Definitions.GetID(sigEventName)
+		if !ok {
+			logger.Debugw("removing signature from events to trace", "signature", sigEventName, "reason", "ID does not exist in definitions")
+			t.config.EngineConfig.Signatures = append(t.config.EngineConfig.Signatures[:i], t.config.EngineConfig.Signatures[i+1:]...)
```

Comment:
I don't think it's a good idea to remove signatures from the engine.
Soon we won't to be able to update the selected events in runtime, and removing them from here will require to reiterate the directories to find signatures and initilization.

Reply:
Then this means we need two lists, one which we track on the side with all currently available signatures, which will allow us to load them later into the engine if they were unloaded. Then there is this list which i'm updating which is the initial signature list going on.
But that external list should probably be added alongside the API work on Loading and Unloading events.